spec/lib/kapost_deploy/plugins/notify_honeybadger_after_promote_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "climate_control"
+require "kapost_deploy/plugins/notify_honeybadger_after_promote"
+
+RSpec.describe KapostDeploy::Plugins::NotifyHoneyBadgerAfterPromote do
+  let(:git_config) { { github_repo: "kapost/kapost_deploy", path: "." } }
+  let(:ahead_releases_double) { double("ahead releases", latest_deploy_version: "1234abc") }
+
+  subject { described_class.new(config, ahead_releases: ahead_releases_double) }
+
+  let(:config) do
+    KapostDeploy::Task.define(:test) do |config|
+      config.app = "scaryskulls-democ"
+      config.to = "scaryskulls-prodc"
+      config.pipeline = "scaryskulls"
+      config.heroku_api_token = "123"
+
+      config.options = { git_config: git_config }
+    end
+  end
+
+  context "when CIRCLE_BUILD_NUM enviroment is not available" do
+    it "notifies honeybadger after promote using pipeline sha" do
```

Comment:
:art: What if, for these two specs, `Kernel` was injected via the initializer? Then you could use a class spy for it and do something like this:

```
subject.after
expect(shell).to have_received(<honeybadger CLI>)
```

`shell` in this case would be something like this:

```
def initialize shell: Kernel
  @shell = shell
  ...
end
```